generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/ArtifactResolver.java

```diff
@@ -0,0 +1,83 @@
+package com.google.devtools.build.workspace.maven;
+
+
+import com.google.common.collect.ImmutableList;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyFilter;
+import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.eclipse.aether.resolution.DependencyResult;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.filter.DependencyFilterUtils;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.function.Function;
+import java.util.logging.Logger;
+
+import static com.google.devtools.build.workspace.maven.AetherUtils.*;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Resolves transitive dependencies of maven artifacts using aether.
+ */
+public class ArtifactResolver {
+  private final RepositorySystem system;
+  private final RepositorySystemSession systemSession;
+  private final List<RemoteRepository> remoteRepositories;
+  private final List<Dependency> managedDependencies;
+
+  private final static Logger logger = Logger.getLogger(
+      MethodHandles.lookup().lookupClass().getName());
+
+  //TODO(petros): add support for managed dependencies, exclusions, and aliases.
+  public ArtifactResolver() {
+    this.system = newRepositorySystem();
+    this.systemSession = newRepositorySession(this.system);
+    this.remoteRepositories = ImmutableList.of(mavenCentralRepository());
+    this.managedDependencies = ImmutableList.of();
+  }
+
+  /**
+   * Given a set of coordinates, resolves the transitive dependencies, and then returns the root
+   * node to the resolved dependency graph. The root node is a sentinel node with direct edges
+   * on the artifacts users declared explicit on.
+   */
+  public DependencyNode resolveArtifacts(List<String> artifacts) {
+    DependencyRequest dependencyRequest = createDependencyRequest(artifacts);
+    DependencyResult dependencyResult;
+
+    try {
+      dependencyResult = system.resolveDependencies(systemSession, dependencyRequest);
+    } catch (DependencyResolutionException e) {
+      //FIXME(petros): This is very fragile. If even one artifact does not resolve, then no artifacts resolve.
```

Comment:
Little change. I return `dependencyResult.getRoot()` and `dependencyResult` is assigned within the try block. I would need to at some point, return null.